src/ui.rs

```diff
@@ -456,6 +457,7 @@ impl UiContext {
 	}
 
 	const fn id_from_index(index: usize) -> WidgetId {
+		// SAFETY: The system cannot access more memory than usize, therefore a usize bound to a number of items cannot overflow.
```

Comment:
hmmmm not sure about how clear this comment is. I think I'll let it be but I feel like the comment on the NPO above is sufficient

Reply:
i think its good since i had a moment where i kind of had the urge to handle an index of `usize::MAX`. But since the index is bound by a number of things in system memory, it is constrained to the system memory, which is constrained to `usize::MAX` already

Reply:
When you say "bound by a number of things", you mean that it can't overflow because the size of a widget is more than 1 byte, right? I think then it's better to say that it's because it's an index on multi-byte items.

(That being said, `usize::MAX` isn't even the maximum value of a pointer, since a memory address on a 64-bit machine actually only uses 48 bits. Not sure about 32-bit machines.)

Although I don't think I'll ever have more than 10000 widgets on the screen for any application I might make with this kind of system :p

Reply:
i actually wrote
> a usize bound to a number of items

which i think is a bit clearer.

> (That being said, usize::MAX isn't even the maximum value of a pointer, since a memory address on a 64-bit machine actually only uses 48 bits. Not sure about 32-bit machines.)
that means on 64bit it CANNOT overflow at all due to n(objects in ram) > bytes(ram) is impossible, but it might be on 32-bit (if thats also 48-bit addressing).
but a widget is a significant number of bytes long, meaning you can multiply it by, say, 16 and boom you can't have more ids than ram: 16 * n(objects) > bytes(ram) is basically impossible unless we somehow come across a 16bit system that can run GUI apps and with gigabytes of ram (not happening).

Reply:
Alright I think it's fine now